### PR TITLE
Ports all remaining interop tests from zipkin-scala 1.38 to java

### LINKED ISSUE #110: Port all interop tests from zipkin-scala 1.38 to java
We've ported some tests from scala to java, as many maintainers cannot debug them otherwise. We are still missing some important tests, such as ensuring span.duration is exported by query time

See https://github.com/openzipkin/zipkin/pull/1057

### PR DESCRIPTION
**Pull request by adriancole**
This is the last set of interop tests to port, particularly around
the dependency link functionality.

Fixes #110

### CODE REVIEW

**Comment by adriancole** [comment]
cc @anuraaga @marcingrzejszczak the scala test still run, but the java ones should fail for the same reasons, and be easier to debug

File: zipkin/src/test/java/zipkin/DependenciesTest.java [hunk 1]
```diff
@@ -41,35 +47,330 @@
...
+            .addAnnotation(Annotation.create(1445136540397049L, CLIENT_RECV, twoPort3002))
+            .build()
+    );
+    processDependencies(trace);
+
+    long traceDuration = trace.get(0).duration;
+
+    assertThat(
+        store.getDependencies((trace.get(0).timestamp + traceDuration) / 1000, traceDuration / 1000)
+    ).containsExactly(
```
Comment (anuraaga) [inline zipkin/src/test/java/zipkin/DependenciesTest.java]: The scala version uses theSameElementsAs, which doesn't check order while this PR uses containsExactly which does. Should we change these to containsOnly?

Comment (anuraaga) [inline zipkin/src/test/java/zipkin/DependenciesTest.java]: For reference, scala
https://github.com/openzipkin/zipkin/blob/79161d905f5aafdfa3aa7d8c4a1acb5ddc69d66f/zipkin-common/src/test/scala/com/twitter/zipkin/storage/DependencyStoreSpec.scala#L128

File: zipkin/src/test/java/zipkin/DependenciesTest.java [hunk 2]
```diff
@@ -41,35 +47,330 @@
...
+
+  /**
+   * The primary annotation used in the dependency graph is [[Constants.SERVER_RECV]]
+   */
+  @Test
+  public void getDependenciesMultiLevel() {
+    processDependencies(trace);
+
+    assertThat(store.getDependencies(today + 1000L, null))
+        .isEqualTo(dep.links);
```
Comment (anuraaga) [inline zipkin/src/test/java/zipkin/DependenciesTest.java]: I think isEqualTo has the same problem

File: zipkin/src/test/java/zipkin/DependenciesTest.java (hunk 1 above)
Comment (adriancole) [inline zipkin/src/test/java/zipkin/DependenciesTest.java]: afaik there's no documented order for links, so feel free to relax to
contains only.

**Pull request merged**